modules/admin-top-bar/assets/js/admin.js

```diff
@@ -0,0 +1,25 @@
+import AdminTopBar from './admin-top-bar';
+
+const AppWrapper = elementorCommon.config.isDebug ? React.StrictMode : React.Fragment;
+document.addEventListener( 'DOMContentLoaded', () => {
+	const isTopBarOptionWidgetChecked = !! document.querySelector( '#e-dashboard-widget-admin-top-bar-hide' );
+	const elementorMenuItemIds = [
+		'toplevel_page_elementor',
+		'menu-posts-elementor_library',
+	];
+
+	const menuItemSelector = elementorMenuItemIds
+		.map( ( itemId ) => `#${ itemId } .wp-menu-open` )
+		.join( ', ' );
+
+	const isElementorPage = !! document.querySelector( menuItemSelector );
+
+	if ( isElementorPage || isTopBarOptionWidgetChecked ) {
+		ReactDOM.render(
+			<AppWrapper>
+				<AdminTopBar isDashboard={isTopBarOptionWidgetChecked} />
```

Comment:
There is leftover code in JS and maybe CSS. after talking with Aviad, he decided to open another task for that and merge this code as fast as we can.